### DISCUSSION TITLE
Achieve concurrency with reactive messaging SQS

### DISCUSSION THREAD

**Discussion by prakashelango** [smallrye/smallrye-reactive-messaging#2636] [Q&A]
As we use SQS reactive messaging

1. [docs shows ](http://smallrye.io/smallrye-reactive-messaging/4.21.0/concepts/incoming-concurrency/) as experimental does this applies for only Kafka or for all reactive messaging (SQS)?
2. Is there a way to achieve concurrency if it's an experimental feature any other alternate suggestions
3. SQS reactive messaging is in preview is that good to use in prod?

![Pasted Graphic 1](https://github.com/smallrye/smallrye-reactive-messaging/assets/28904745/75531c63-af6c-4e56-9660-33d421e4e81f)

**Comment by ozangunalp** [comment]
The SQS connector is in preview because it lacks some tests we may have on other connectors. 
For example the concurrency feature (which is not Kafka specific) that you mentioned should work, but we don't have the tests for it.
This is where the community feedback and contributions are very valuable.

Another way to process messages concurrently is using [`@Blocking(ordered = false)`](https://smallrye.io/smallrye-reactive-messaging/latest/concepts/blocking/)

**Reply by prakashelango** [reply]
I see, when we leverage blocking along with virtual threads do we need to set executor pool concurrency? 

Do you mind pointing to tests let’s me give an attempt to add it.

**Reply by ozangunalp** [reply]
It'll look something like https://github.com/smallrye/smallrye-reactive-messaging/blob/main/smallrye-reactive-messaging-amqp/src/test/java/io/smallrye/reactive/messaging/amqp/ConcurrentProcessorTest.java

**Reply by ozangunalp** [reply]
For virtual-threads usage you can refer to https://quarkus.io/guides/messaging#execution-model or https://quarkus.io/guides/messaging-virtual-threads
